src/Fantomas.CoreGlobalTool/Program.fs

```diff
@@ -313,23 +331,90 @@ let main argv =
             let version = CodeFormatter.GetVersion()
             printfn "Fantomas v%s" version
         else
-            match inputPath, outputPath with
-            | InputPath.Unspecified, _ ->
-                eprintfn "Input path is missing..."
-                exit 1
-            | InputPath.Folder p1, OutputPath.Notknown -> processFolder p1 p1
-            | InputPath.File p1, OutputPath.Notknown -> processFile p1 p1 config
-            | InputPath.File p1, OutputPath.IO p2 ->
-                processFile p1 p2 config
-            | InputPath.Folder p1, OutputPath.IO p2 -> processFolder p1 p2
-            | InputPath.StdIn s, OutputPath.IO p ->
-                stringToFile s p config
-            | InputPath.StdIn s, OutputPath.Notknown
-            | InputPath.StdIn s, OutputPath.StdOut ->
-                stringToStdOut s config
-            | InputPath.File p, OutputPath.StdOut ->
-                fileToStdOut p config
-            | InputPath.Folder p, OutputPath.StdOut ->
-                allFiles recurse p
-                |> Seq.iter (fun p -> fileToStdOut p config)
+            let check = results.Contains<@ Arguments.Check @>
+
+            let checkFile config file =
+                FakeHelpers.formatFileAsync config file
+
+            let toCheckOutput r =
+                match r with
+                | FakeHelpers.FormatResult.Formatted(filename, _) -> sprintf "%s needs formatting" filename
+                | FakeHelpers.FormatResult.Error(filename, exn) -> sprintf "error: Failed to format %s: %s" filename (exn.ToString())
+                | FakeHelpers.FormatResult.Unchanged(_) -> failwith "No unchanged file should be present at this point"
+
+            let reportCheckResults (output: TextWriter) (results: seq<FakeHelpers.FormatResult>) =
+                results
+                |> Seq.map toCheckOutput
+                |> Seq.iter output.WriteLine
+
+            let runCheck (filenames: seq<string>) =
+                async {
+                    let! formatted =
+                        filenames
+                        |> Seq.map (checkFile config)
+                        |> Async.Parallel
+
+                    let isChange =
+                        function
+                        | FakeHelpers.FormatResult.Unchanged(_) -> false
+                        | FakeHelpers.FormatResult.Formatted(_) | FakeHelpers.FormatResult.Error(_) -> true
+
+                    let changes = formatted |> Seq.filter isChange
+
+                    // always print report to StdOut
+                    changes |> reportCheckResults stdout
+
+                    let isError =
+                        function
+                        | FakeHelpers.FormatResult.Error(_) -> true
+                        | _ -> false
+
+                    let hasErrors =
+                        changes
+                        |> Seq.exists isError
+
+                    if hasErrors then
+                        return 99
```

Comment:
Any particular reason you went with 99? Just wondering though.

Reply:
Ahah. Not really. We could go with your lucky number. Black (the python formatter) chose 123.

Reply:
I was secretly hoping it was [a Jay-Z reference](https://open.spotify.com/track/7IdFdRlCjUi6kkhbPoRfnw?si=Qb3yoi0zR2eAhQC2VDYpag) 😋

@jindraivanek do you have a lucky number? I'm ok with 99 or anything else that sense.